src/fsw/FCCode/OrbitController.cpp

```diff
@@ -120,31 +163,55 @@ void OrbitController::execute() {
 }
 
 double OrbitController::time_till_node(double theta, const lin::Vector3d &pos, const lin::Vector3d &vel) {
+
+
+    // Collect Relative Orbit Estimator State
+    unsigned char rel_orbit_state=rel_orbit_valid_fp->get();
+
     // Calculate angular velocity (w = v/r)
     double ang_vel = lin::norm(vel)/lin::norm(pos);
 
     // Calculate the times until each node (theta_node = theta_now + w*t)
-    double min_time = std::numeric_limits<double>::max();
-    for (size_t i=0; i < 3; i++) {
-        double time_til_node = (firing_nodes[i] - theta) / ang_vel;
-        if (time_til_node > 0 && time_til_node < min_time) {
-            min_time = time_til_node;
+    auto next_node = [&](auto const &arr) -> double {
+        auto min_time = std::numeric_limits<double>::max();
+        for (auto const node : arr) {
+            auto const time_til_node = (node - theta) / ang_vel;
+            if (time_til_node > 0 && time_til_node < min_time) {
+                min_time = time_til_node;
+            }
         }
-    }
+        return min_time;
+    };
 
-    // Return the smallest positive time
-    return min_time;
+    return (rel_orbit_state == static_cast<unsigned char>(rel_orbit_state_t::estimating)) ? 
+            next_node(firing_nodes_near) : next_node(firing_nodes_far);
 }
 
 lin::Vector3d OrbitController::calculate_impulse(double t, const lin::Vector3d &r, const lin::Vector3d &v, 
-    const lin::Vector3d &dr, const lin::Vector3d &dv) {
+    const lin::Vector3d &dr_smoothed, const lin::Vector3d &dv_smoothed) {
```

Comment:
I think we should change the name of the input parameters just back to `dr` `dv` here as the function doesn't really care if it's handling smoothed data or not.
```suggestion
    const lin::Vector3d &dr, const lin::Vector3d &dv) {
```